pkg/migration/migrationmanager/manager.go

```diff
@@ -78,38 +102,14 @@ func (m *Manager) Migrate(ctx context.Context, from, to clusterversion.ClusterVe
 		return nil
 	}
 
-	// TODO(irfansharif): We'll need to acquire a lease here and refresh it
-	// throughout during the migration to ensure mutual exclusion.
-
-	// TODO(irfansharif): We'll need to create a system table to store
-	// in-progress state of long running migrations, for introspection.
-
-	clusterVersions := clusterversion.ListBetween(from, to)
-	if len(clusterVersions) == 0 {
-		// We're attempt to migrate to something that's not defined in cluster
-		// versions. This only happens in tests, when we're exercising version
-		// upgrades over non-existent versions (like in the cluster_version
-		// logictest). These tests explicitly override the
-		// binary{,MinSupportedVersion} in order to work. End-user attempts to
-		// do something similar would be caught at the sql layer (also tested in
-		// the same logictest). We'll just explicitly append the target version
-		// here instead, so that we're able to actually migrate into it.
-		clusterVersions = append(clusterVersions, to)
-	}
+	clusterVersions := m.listBetween(from, to)
 	log.Infof(ctx, "migrating cluster from %s to %s (stepping through %s)", from, to, clusterVersions)
 
 	for _, clusterVersion := range clusterVersions {
-		cluster := newCluster(m.nl, m.dialer, m.executor, m.db)
-		h := newHelper(cluster, clusterVersion)
-
-		// First run the actual migration (if any). The cluster version bump
-		// will be rolled out afterwards. This lets us provide the invariant
-		// that if a version=V is active, all data is guaranteed to have
-		// migrated.
-		if migration, ok := registry[clusterVersion]; ok {
-			if err := migration.Run(ctx, h); err != nil {
-				return err
-			}
+		log.Infof(ctx, "stepping through %s to %s (%s)", from, to, clusterVersions)
```

Comment:
Do we want to log the specific `clusterVersion` instead of the list of _all_ cluster versions?

Reply:
I think I just fixed this.